### PR TITLE
DropDownEditor: Detach focus events before attaching new ones (T1238121)

### PR DESCRIPTION
**Pull request by marker-dao**

### CODE REVIEW

File: packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/dropDownEditor.tests.js [hunk 1]
```diff
@@ -457,6 +457,66 @@ QUnit.module('focus policy', () => {
         assert.ok($dropDownEditor.hasClass('dx-state-focused'), 'Widget is focused after click on clearButton');
     });
 
+    QUnit.test('editor should fire onFocusIn once, if fieldTemplate is used', function(assert) {
```
Comment (ksercs) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/dropDownEditor.tests.js]:
```suggestion
    QUnit.test('editor should fire onFocusIn once if fieldTemplate is used (T1238121)', function(assert) {
```

File: packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/dropDownEditor.tests.js [hunk 2]
```diff
@@ -457,6 +457,66 @@ QUnit.module('focus policy', () => {
...
+
+        const $dropDownEditor = $('#dropDownEditorLazy').dxDropDownEditor({
+            items: [
+                { name: 'one', id: 1 },
+                { name: 'two', id: 2 },
+                { name: 'three', id: 3 },
+            ],
+            displayExpr: 'name',
+            valueExpr: 'id',
+            value: 1,
```
Comment (ksercs) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/dropDownEditor.tests.js]: Cannot we minimize the test? Now a lot of properties looks excess

File: packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/dropDownEditor.tests.js (hunk 1 above)
Comment (marker-dao) [reply packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/dropDownEditor.tests.js]: [7894ae7f11962bb566873551ef637e5348c23f52]

File: packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/dropDownEditor.tests.js (hunk 2 above)
Comment (marker-dao) [reply packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/dropDownEditor.tests.js]: [7894ae7f11962bb566873551ef637e5348c23f52]

**Review by Zedwag — APPROVED** [review]

**Review by ksercs — APPROVED** [review]

**Pull request merged**